docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -63,6 +63,10 @@ Some organizations require selecting hardening rules. For example, the US govern
 
 [^CMU2018]: Carnegie Mellon University (CMU), [Top 10 Secure Coding Practices](https://wiki.sei.cmu.edu/confluence/display/seccode/Top+10+Secure+Coding+Practices), SEI CERT Coding Standards Wiki, 2018-05-02.
 
+**What should you do when compiling compilers?**
+
+If you are compiling a C/C++ compiler, where practical make the generated compiler's default options the *secure* options (as documented here). For example, when compiling GCC, use `--enable-default-pie` (which enables the flags `-fPIE` and `-pie` by default when using the generated compiler executable) and when compiling clang on Linux systems set `CLANG_DEFAULT_PIE_ON_LINUX` (which has a similar effect).
```

Comment:
nit-pick: when reading the text in the parenthesis, I initially expect the "here" to be a link instead of referring to this paragraph.

```suggestion
If you are compiling a C/C++ compiler, where practical make the generated compiler's default options the *secure* options. For example, when compiling GCC, use `--enable-default-pie` (which enables the flags `-fPIE` and `-pie` by default when using the generated compiler executable) and when compiling clang on Linux systems set `CLANG_DEFAULT_PIE_ON_LINUX` (which has a similar effect).
```

Reply:
Clang would like to deprecate that CMake option and is instead pushing people towards Clang configuration files.

You can see what we do in Gentoo at https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-devel/clang-common/clang-common-18.0.0_pre20231025.ebuild?id=b162a825ab14ae965ca012599f5efe9283ff7c36#n131.

Reply:
Okay, I'll do that.